### ISSUE TITLE
[no-negated-async]: false positives with equality report

### ISSUE THREAD

**Issue by rafaelss95** [angular-eslint/angular-eslint#315]
When I have a template like this:

```HTML
<div *ngIf="(item$ | async) == 1">{{ item }}</div>
```

...with this configuration:

```
{
  "extends": [
    "plugin:@angular-eslint/template/recommended"
  ],
  "files": ["*.html"],
  "rules": {}
}
```

...I receive the following error message:

> Async pipes must use strict equality `===` when comparing with `false`

...that comes from here:

 https://github.com/angular-eslint/angular-eslint/blob/c8dd14c95f881593d7b6ca9f7ba80b37b7eaf40f/packages/eslint-plugin-template/src/rules/no-negated-async.ts#L25

...which is obviously incorrect, but what intrigues me even more is why we report the loose equality as IMO it's outside the scope of this rule. In any case, you may have more context on why we report this.

The possible fixes that I can see are:
- Report only `false` as the message indicates (but why only `false` and not `== null` or `== undefined` per example?);
- Report all falsy values (`false`, `0`, `-0`, `''`, `null`, `undefined`) and update message accordingly;
- Drop the equality report;

My choice would be the drop of the equality report, however as this rule is included in the _recommended_ config and also @mgechev mentioned sometime ago that they've introduced it in Google codebase I want to hear your opinion about this before touching in any code.

If the drop is a consesus, we could implement something like [**`eqeqeq`**](https://eslint.org/docs/rules/eqeqeq)/[**`triple-equals`**](https://palantir.github.io/tslint/rules/triple-equals) for templates ([that I've suggested in Codelyzer](https://github.com/mgechev/codelyzer/issues/643)) to handle the equality in a more general way.

cc @JamesHenry @mgechev

**Comment by allout58** [comment]
I think at a bare minimum the rules should be split out. Personally, I don't care too much about strict equality for falsy values, but don't want to allow negated async pipes, so I'd like to be able to disable part of this rule as it stands now without disabling the other part.

**Comment by ibedard16** [comment]
+1 to splitting the rules. Granularity should definitely be preferred for linter rules.

I'm definitely in favor of implementing a rule like `[eqeqeq](https://eslint.org/docs/rules/eqeqeq)` or `[triple-equals](https://palantir.github.io/tslint/rules/triple-equals)`, especially since the rule can be customized to allow for null checks.

**Comment by JamesHenry** [comment]
@rafaelss95 Now that the rule has been split how, what would you like to do with this issue?

**Comment by rafaelss95** [comment]
IMO it can be closed now :)

**Comment by JamesHenry** [comment]
Cool, thanks!
